src/components/Create/components/RewardsList/AddEditRewardModal.tsx

```diff
@@ -122,15 +148,35 @@ export const AddEditRewardModal = ({
             supportedFileTypes={
               new Set(['image/jpeg', 'image/png', 'image/gif'])
             }
+            beforeUpload={onBeforeUpload}
             customRequest={onCustomRequest}
             listType="picture-card" // Tried to do away with styling, but need this -.-
           >
-            {({ uploadUrl, isUploading, undo }) => {
+            {({ uploadUrl, isUploading, undo, percent }) => {
               if (isUploading) {
-                return <LoadingOutlined />
+                return (
+                  <div>
+                    <Progress
+                      width={48}
+                      className="h-8 w-8"
+                      strokeColor={colors.background.action.primary}
+                      type="circle"
+                      percent={percent}
+                      format={percent => (
+                        <div className="text-black dark:text-grey-200">
+                          {percent ?? 0}%
+                        </div>
+                      )}
+                    />
+                  </div>
+                )
               }
               if (uploadUrl === undefined) {
-                return <UploadButton />
+                return (
+                  <>
+                    <UploadButton />
+                  </>
```

Comment:
any reason for this diff?
```suggestion
            <UploadButton />
```